### PR TITLE
Add support for get filtered list of extensions

### PR DESCRIPTION
**Pull request by sdkrach**
Enhanced the existing logic to be able to query multiple extensions by a filter attribute. Subsequenty, simplified the single extension code to reuse the multiple extension method.

### CODE REVIEW

File: bundles/org.palladiosimulator.commons/src/org/palladiosimulator/commons/eclipseutils/ExtensionHelper.java [hunk 1]
```diff
@@ -176,32 +176,69 @@ private ExtensionHelper() {
...
 
         for (final IExtension extension : extensions) {
             final IConfigurationElement configurationElement = obtainConfigurationElement(extension, elementName);
 
-            if (configurationElement.getAttribute(filterAttributeName).equals(filterAttributeValue)) {
+            if (configurationElement.getAttribute(filterAttributeName)
+                .equals(filterAttributeValue)) {
                 try {
                     @SuppressWarnings("unchecked")
                     final DATA_TYPE executableExtension = (DATA_TYPE) configurationElement
```
Comment (seiferma) [inline bundles/org.palladiosimulator.commons/src/org/palladiosimulator/commons/eclipseutils/ExtensionHelper.java:197]: I cannot find a check before this cast. Wouldn't it be better to test for the type and throw an exception if the type does not match?

**Review by seiferma — CHANGES_REQUESTED** [review]
Looks good to me but it might be good to add an instance check before the cast.

File: bundles/org.palladiosimulator.commons/src/org/palladiosimulator/commons/eclipseutils/ExtensionHelper.java (hunk 1 above)
Comment (sdkrach) [reply bundles/org.palladiosimulator.commons/src/org/palladiosimulator/commons/eclipseutils/ExtensionHelper.java:197]: The cast is done based on the generic type parameter. There is no way of type checking that. We would need to adapt the signature to include an additional class object parameter. I would rather omit that.

Comment (sdkrach) [reply bundles/org.palladiosimulator.commons/src/org/palladiosimulator/commons/eclipseutils/ExtensionHelper.java]: Added additional signatures which allow safe-casting and deprecated the current single extension case.

**Review by seiferma — APPROVED** [review]
Great, thanks.

**Pull request merged**